### ISSUE TITLE
Possibility of panoptic segmentation

### ISSUE THREAD

**Issue by auto-enthu** [dbolya/yolact#285]
Hi @dbolya,
I have a few queries:

1. Sorry if I sound a bit naive. I am not clear from your paper what do you exactly mean by "Mask coefficient" or how is this measured. I cannot understand how to calculate this coefficient value or how "the third branch .. predicts k mask coefficients" (Section 3.2)

2. What are the possibilities to include panoptic segmentation to the project?
I wonder if we can include a full image fcn segmentation to the Yolact as a head. [[Link](https://arxiv.org/pdf/1801.00868.pdf)]

**Comment by dbolya** [comment]
1. The mask coefficients are just the output of a conv layer, the same way the bounding box regression coefficients and class confidence values are. It's not like they're fixed or anything, you can think of them as weights to apply to the prototypes when combining them.
2. I think panoptic segmentation using the YOLACT pipeline shouldn't be difficult at all, but it would be rather tricky to seat it into the code as it is now. I'd like to do that as a future project, but it's not high on the priority list right now.

**Comment by abhigoku10** [comment]
@auto-enthu  @dbolya  to obtain panoptic segmentation what are the changes have to made ??

**Comment by dbolya** [comment]
@abhigoku10 You essentially need to add an additional semantic segmentation pipeline to deal with the stuff annotations. That in itself is simple because we already have a hidden semantic segmentation branch so you could just beef that up and use that.

The more difficult part is getting all the data loading and evaluation code to support those stuff annotations and detections.

**Comment by abhigoku10** [comment]
@dbolya  i want to train yolact for panoptic segmentation , as u had mentioned the challenging part is to modifiy the data loading and evaluation code right , can you share some pointers on this or elaborate i want to train yolact/yolact++ on cityscapes dataset so

**Comment by dbolya** [comment]
@abhigoku10 I mean there are a lot of modifications that need to be done. I'd start with getting the data loaded and able to be trained on, then training on that, and passing the segmentation through the eval code and use some external evaluation script. I can't help much more than that without partially implementing that myself.

**Comment by abhigoku10** [comment]
@dbolya i was asking this becuase , i wanted to detect stationary objects like trees , builidings ,benches and annotations are available only in the form of semantic segmentation maps or panoptic segmentation maps so thought of modifying the training pipeline . R you aware of any instance segmenation datasets for trees building benches poles like that scenes

**Comment by dbolya** [comment]
@abhigoku10 Yeah I think most datasets have those things annotated as a semantic segmentation since they're pretty hard to annotate per instance.

**Comment by abhigoku10** [comment]
@dbolya can i train cityscapes / camvid semattic segmentation dataset on yolac++ or yolact what will be the output result expected , since instead of passing instance data i am passing semanttic data so ..
